esp/services/ws_access/ws_accessService.cpp

```diff
@@ -241,9 +240,12 @@ void Cws_accessEx::getBasednReq(IEspContext &context, const char* name, const ch
 void Cws_accessEx::setBasedns(IEspContext &context)
 {
     CLdapSecManager* secmgr = (CLdapSecManager*)(context.querySecManager());
+    if (secmgr == nullptr)
```

Comment:
I know the line is not new, but since querySecManager() may return a different type of secmgr than LDAP I think it should be a dynamic_cast.  An old fashion cast or a static_cast may not detect that the manager is not compatible with ldap.
CLdapSecManager* secmgr = (CLdapSecManager*)(context.querySecManager());

CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());

Reply:
@afishbeck Or better yet, call the new ISecObject::checkFeatures to check the level of Security Manager support (avoiding the peculiar dynamic_cast model).  But this should probably all be refactored in a single, separate PR

Reply:
Will change to: CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
The same change has to be in many other places.

Reply:
@wangkx I believe this will result in the message "Security manager is not found. Please check if the system authentication is set up correctly"  if there is a security manager, but it is not LDAP.  This is wrong, perhaps an improved error message stating something like "Security manager is not found, or is not LDAP. Please check if the system authentication is set up correctly"

Reply:
I may also combine the code into the checkUser().